src/cryptonote_core/blockchain.cpp

```diff
@@ -5357,24 +5367,31 @@ bool Blockchain::prepare_handle_incoming_blocks(const std::vector<block_complete
       its = m_scan_table.find(tx_prefix_hash);
       assert(its != m_scan_table.end());
 
+      MINFO(__FILE__ << ":" << __LINE__ << " TODO sean remove this - debug");
       // get all amounts from tx.vin(s)
       for (const auto &txin : tx.vin)
       {
-        const auto& in_to_key = var::get<txin_to_key>(txin);
 
-        // check for duplicate
-        auto it = its->second.find(in_to_key.k_image);
-        if (it != its->second.end())
-          SCAN_TABLE_QUIT("Duplicate key_image found from incoming blocks.");
+        if (!std::holds_alternative<txin_gen>(txin))
```

Comment:
shouldn't this be `if(std::holds_alternative<tx_to_key>(txin))` ?

Reply:
Yeah it should, and there are a few times I have done this. My reasoning is that some of these code blocks (maybe not this one) have `assert input = <txin_to_key>` and wanted those asserts to still trigger if a `txin_to_script` or something weird came through. I don't think `txin_to_scripts` are used here but was unsure, so temporarily just allowing `txin_gen` to exist here with a mental note to lock down what types of inputs can go into a transaction.